include/pushmi/deferred.h

```diff
@@ -114,8 +114,8 @@ class deferred<SF> {
 
 template <PUSHMI_TYPE_CONSTRAINT(Sender<is_none<>>) Data, class DSF>
 class deferred<Data, DSF> {
-  Data data_{};
-  DSF sf_{};
+  Data data_;
+  DSF sf_;
```

Comment:
I added these in-class initializers because I convinced myself that some constructors could potentially leave these uninitialized.

Reply:
the cuda compiler choked. I can Macroize them, or supply explicit default constructors instead. When I was looking at these it appeared to me that they could not be uninitialized. 1. because Data must be a class that will be default constructed. I concede that DSF might not be. 2. because we promote all the non-default construction to the constructor that explicitly initializes all of them.

Reply:
I don't get it. CUDA doesn't seem to mind in-class initialization: https://godbolt.org/g/oDqCBd. What about these cases is giving CUDA a hard time, I wonder.

Reply:
https://raw.githubusercontent.com/kirkshoop/pushmi/cuda/include/pushmi.h(2745): error: the default constructor of "pushmi::detail::constrained_fn<lambda [](auto) mutable->auto, pushmi::NoneReceiverConcept::Eval<pushmi::_1, std::__exception_ptr::exception_ptr>>" cannot be referenced -- it is a deleted function

          detected during instantiation of class "pushmi::deferred<SF> [with SF=pushmi::detail::constrained_fn<lambda [](auto) mutable->auto, pushmi::NoneReceiverConcept::Eval<pushmi::_1, std::__exception_ptr::exception_ptr>>]" 

from [godbolt](https://godbolt.org/g/ENs4t2)